foundation/src/main/java/org/researchstack/foundation/components/presentation/TaskPresentationFragment.kt

```diff
@@ -0,0 +1,241 @@
+package org.researchstack.foundation.components.presentation
+
+import android.content.Context
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.support.v7.app.AlertDialog
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.Toolbar
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import org.researchstack.foundation.R
+import org.researchstack.foundation.components.common.ui.callbacks.StepCallbacks
+import org.researchstack.foundation.components.presentation.interfaces.IStepFragment
+import org.researchstack.foundation.components.presentation.interfaces.IStepFragmentProvider
+import org.researchstack.foundation.components.presentation.interfaces.ITaskNavigator
+import org.researchstack.foundation.core.interfaces.IResult
+import org.researchstack.foundation.core.interfaces.IStep
+import org.researchstack.foundation.core.interfaces.ITask
+import java.lang.RuntimeException
+import java.util.*
+
+abstract class TaskPresentationFragment<StepType: IStep, ResultType: IResult, TaskType: ITask>(): Fragment(), StepCallbacks {
```

Comment:
I'm not sure what you mean by fragments not being directly instantiated.

RE: generic presentation fragment. When we were just using the interface types, it was requiring quite a bit of casting to support backwards compatibility. If you truly want the base interfaces, you should be able to specialize w/ just IStep, IResult, & ITask and it will allow any subtypes of those. However, making this generic allows us to rely on compiler time type checking instead of getting run time errors from explicit casting.